src/main/java/edu/harvard/iq/dataverse/api/SavedSearches.java

```diff
@@ -172,17 +210,31 @@ public Response add(JsonObject body) {
     }
 
     @DELETE
+    @AuthRequired
     @Path("{id}")
-    public Response delete(@PathParam("id") long doomedId) {
-        boolean disabled = true;
-        if (disabled) {
-            return error(BAD_REQUEST, "Saved Searches can not safely be deleted because links can not safely be deleted. See https://github.com/IQSS/dataverse/issues/1364 for details.");
+    public Response delete(@Context ContainerRequestContext crc, @PathParam("id") long doomedId, @QueryParam("unlink") boolean unlink) {
+
+        AuthenticatedUser superuser = null;
+        try {
+            superuser = getRequestAuthenticatedUserOrDie(crc);
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
         }
+        if (superuser == null || !superuser.isSuperuser()) {
+            return error(Response.Status.UNAUTHORIZED, "Superusers only.");
+        }
+
         SavedSearch doomed = savedSearchSvc.find(doomedId);
         if (doomed == null) {
             return error(NOT_FOUND, "Could not find saved search id " + doomedId);
         }
-        boolean wasDeleted = savedSearchSvc.delete(doomedId);
+        boolean wasDeleted;
+        try {
+            wasDeleted = savedSearchSvc.delete(doomedId, unlink);
+        } catch (Exception e) {
+            return error(INTERNAL_SERVER_ERROR, "Problem while trying to unlink links of saved search id " + doomedId);
```

Comment:
Should we return the exception so the user has an idea of what went wrong?

Reply:
I don't think it's a good practice to show exception in json API return (security / readability). 
Also, as the this is "`api/admin/savedsearches`, you are in localhost so you can check the logs.